### ISSUE TITLE
Acceptance tests leak Container on every test run.

### ISSUE THREAD

**Issue by julkiewicz** [emberjs/ember.js#12490]
I came across this issue in my app. Basically I have a pretty large test suite and it hits Chrome tab memory limit. Clearly this is unacceptable, and prevents me from running tests on my integration server.

Here is a JS Bin repro:

http://jsbin.com/tudalurebo/1/edit?html,js,output

Steps:
1. Open Developer tools in a new tab.
2. Open JSBin under Chrome.
3. Wait for the Developer tools to hit "debugger" line.
4. Take a Memory Snapshot (note, Chrome automatically continues once a snapshot is complete, not sure why that is).
5. Continue 10x to better demonstrate a problem.
6. Take another Memory Snapshot.
7. Use comparison between Snapshot 2 and Snapshot 1 to see that memory is leaking, most notably "Container" objects that hold references to virtually everything else.
8. Scroll to see +10 delta next to "Container".

The following screenshot demonstrates the results on my machine: 

https://files.slack.com/files-pri/T0451CN7C-F0CKSDG2W/zrzut_ekranu_2015-10-16_o_20.44.57.png

**Comment by stefanpenner** [comment]
cannot replicate

**Comment by julkiewicz** [comment]
Actually, let me retry this again. I'd swear I got a leak yesterday when testing this, however perhaps I made a mistake. Let me come back to you when I get this right.

**Comment by stefanpenner** [comment]
@julkiewicz its also possible you noticed the issue with an older version of several libs. I spent time the last few months trying to squash all these leaks. So its possible I already fixed the one you encountered.

**Comment by stefanpenner** [comment]
going to close this, will gladly reopen if a reproduction can be provided

**Comment by julkiewicz** [comment]
I think it was Chrome playing a joke on me. After I restarted it, my own app doesn't leak memory either.

**Comment by huguesdk** [comment]
I’m working on a large Ember.js application and I have the same issue. I’m not using Ember CLI. `App.reset()` is called between each test.

At first I thought it was a PhantomJS problem but it happens also with Firefox. In PhantomJS, with Ember.js 1.9.1 and 570 tests, the memory usage increases by around 500 MiB. With the same tests in Ember.js 1.10.0, it grows by more than 1200 MiB.

Currently, with Ember 2.0 or 2.1, PhantomJS crashes after around 500 tests (out of 1000+) with a memory usage growing up by around 4 GiB.

Any advice on how to debug this?

**Comment by stefanpenner** [comment]
Without a reproduction to work with it is hard to diagnose

**Comment by huguesdk** [comment]
I tried the steps julkiewicz lists, and I have the same results: a new Container after each test. In Chromium 25.0.1364.172, it is always the case. In Google Chrome 46.0.2490.80, it is the case only if taking memory snapshots one after the other and never resuming execution manually. If execution is resumed manually, the number of Container objects falls back to 1.

**Comment by calderas** [comment]
I'm going through something similar on my app after migrating to from ember 1.12 to ember 1.13.10.
I made sure to upgrade all the libs with the latests improvements qunit, ember-qunit, etc.

Following a similar approach as #11748 I wrote something up, although I'm not sure it demonstrates the problem entirely or if its even an issue so far:
https://github.com/calderas/leaky-mcleakerson/issues/2

Hopefully @stefanpenner can take a quick look and suggest some ideas how to debug further.
